File: lyedu-api/src/main/java/com/lyedu/entity/UserTask.java
package com.lyedu.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 用户任务进度实体
 *
 * @author LyEdu Team
 */
@Data
public class UserTask {

    private Long id;
    private Long userId;
    private Long taskId;
    /** 进度 JSON：{"items":[{"type":"course","id":1,"done":1},...]} */
    private String progress;
    /** 状态：0-进行中，1-已完成 */
    private Integer status;
    private LocalDateTime completedAt;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: lyedu-h5/src/api/userTask.ts
import request from '@/utils/request'

export interface UserTask {
  id: number
  userId: number
  taskId: number
  progress?: string
  status: number
  completedAt?: string
  createTime?: string
  updateTime?: string
}

export interface TaskWithUserProgress {
  task: { id: number; title?: string; [key: string]: unknown }
  userTask: UserTask | null
}

export const getMyTasks = () => request.get<TaskWithUserProgress[]>('/user-task/my')

export const getTaskDetail = (taskId: number) =>
  request.get<{ id: number; title?: string; description?: string; [key: string]: unknown }>(`/user-task/task/${taskId}`)
